server/api/services/transaction_service.go

```diff
@@ -62,6 +62,17 @@ func (ts *transactionService) Initialize(userId string, transaction *models.Tran
 		return nil, echo.NewHTTPError(http.StatusNotFound, "unable to find user")
 	}
 
+	foundTransaction, err := ts.transactionPersister.GetByIdentifier(transaction.Identifier, ts.tenant.ID)
+	if err != nil {
+		ts.logger.Error(err)
+		return nil, echo.NewHTTPError(http.StatusInternalServerError, "unable to search for transaction")
+	}
+
+	if foundTransaction != nil {
```

Comment:
```suggestion
	if foundTransaction != nil && len(*foundTransaction) > 0 {
```

`GetByIdentifier` always returns an empty slice when no transaction was found. And only on an error it returns `nil`.